net2/DNSManager.js

```diff
@@ -49,98 +49,93 @@ const DNSQUERYBATCHSIZE=5;
-                     callback(err, null);
-                }
-            });
+          callback(err, null);
         }
+      });
     }
+  }
 
-    // Reslve v6 or v4 address into a local host
-    resolveLocalHost(ip, callback) {
-        if (iptool.isV4Format(ip)) {
-            rclient.hgetall("host:ip4:" + ip, (err, data) => {
-                if (data && data.mac) {
-                    rclient.hgetall("host:mac:" + data.mac, (err, data) => {
-                        if (err == null && data != null) {
-                            callback(err, data);
-                        } else {
-                            callback(err, null);
-                        }
-                    });
-
-                } else {
-                    callback("404", null);
-                }
-            });
-        } else if (iptool.isV6Format(ip)) {
-            rclient.hgetall("host:ip6:" + ip, (err, data) => {
-                if (err == null && data != null) {
-                    if (data.mac) {
-                        rclient.hgetall("host:mac:" + data.mac, (err, data) => {
-                            if (err == null && data != null) {
-                                callback(err, data);
-                            } else {
-                                callback(err, null);
-                            }
-                        });
-                    } else {
-                        callback(null, null);
-                    }
-                } else {
-                    callback(err, null);
-                }
-            });
-        } else {
-            log.error("DNSManager:ResolveHost:Error", ip);
-            callback("bad ip", null);
-        }
-    }
+  // Reslve v6 or v4 address into a local host
+  resolveLocalHost(ip, callback) {
+    callback = callback || function() {}
 
-    findHostWithIP(ip, callback) {
-        let key = "host:ip4:" + ip;
-        log.debug("DNS:FindHostWithIP", key, ip);
-        rclient.hgetall(key, (err, data) => {
-            let mackey = "host:mac:" + data.mac;
-            rclient.hgetall(mackey, (err, data) => {
-                callback(mackey, err, data);
-            });
-        });
+    this.resolveLocalHostAsync(ip)
+       .then(res => callback(null, res))
+       .catch(err => {
+         callback(err);
+       })
+  }
+
+  async resolveLocalHostAsync(ip, callback) {
+    let mac;
+
+    if (iptool.isV4Format(ip)) {
+      let data = await rclient.hgetallAsync("host:ip4:" + ip)
+      if (data && data.mac) {
+        mac = data.mac
+      } else {
+        throw new Error('IP Not Found: ' + ip);
+      }
+    } else if (iptool.isV6Format(ip)) {
+      let data = await rclient.hgetallAsync("host:ip6:" + ip)
+      if (data && data.mac) {
+        mac = data.mac
+      } else {
+        throw new Error('IP Not Found: ' + ip);
+      }
+    } else {
+      log.error("ResolveHost:BadIP", ip);
+      throw new Error('bad ip');
     }
 
-/*
+    return rclient.hgetallAsync("host:mac:" + mac)
```

Comment:
could reuse the functions from HostTool.js

e.g. https://github.com/firewalla/firewalla/blob/master/net2/HostTool.js#L72